finagle-http/src/main/scala/com/twitter/finagle/http/filter/StatsFilter.scala

```diff
@@ -23,6 +23,12 @@ class StatsFilter[REQUEST <: Request](stats: StatsReceiver)
   private[this] val timeReceiver = stats.scope("time")
   private[this] val responseSizeStat = stats.stat("response_size")
 
+  private[this] val counterCache: Map[String, Counter] =
+    Map.empty.withDefault(statusReceiver.counter(_))
```

Comment:
I don't follow.  MemoizedSR decorates another SR, right?  Wouldn't that solve the problem here?  Why not just wrap the underlying SR with a MemoizedSR in the StatsFilter?